lib/allwinner-r8.js

```diff
@@ -102,20 +102,33 @@ AllwinnerR8.prototype.pinMode = function(pin, mode) {
     this._writeRegister(cfgReg, cfgVal);
   }
 
-  if (pin === 'PWM0' && mode === 2) {
-    var pwmCtrlVal = (1 << 6) | (1 << 5) | (1 << 4);
+  if (pin === 'PWM0') {
+    var pwmCtrlVal = 0;
     var pwmPeriodVal = (0xff << 16) | 0;
 
+    if (mode === 2) {
+      pwmCtrlVal = (1 << 6) | (1 << 5) | (1 << 4);
+    }
+
     this._writeRegister(PWM_CH0_PERIOD, pwmPeriodVal);
     this._writeRegister(PWM_CTRL, pwmCtrlVal);
   }
 };
 
 AllwinnerR8.prototype.pwmWrite = function(pin, value) {
   if (pin === 'PWM0') {
-    var pwmPeriodVal = (0xff << 16) | Math.round(value);
+    if (value === 0) {
+      this.pinMode('PWM0', 1);
+      this.digitalWrite('PWM0', 0);
+    } else if (value === 255) {
```

Comment:
Doesn't `led.on()` call analog write with a value of 255?

I think Arduino code also has special checks like this so that 0 maps to LOW and 255 maps to HIGH.